src/PoolManager.sol

```diff
@@ -275,11 +282,11 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         if (currency.isNative()) {
             paid = msg.value;
         } else {
-            uint256 reservesBefore = reservesOf[currency];
-            reservesOf[currency] = currency.balanceOfSelf();
-            paid = reservesOf[currency] - reservesBefore;
+            uint256 reservesBefore = Reserves.get(currency);
+            if (reservesBefore == 0) revert ReservesMustBeSynced();
+            uint256 reservesNow = sync(currency);
```

Comment:
imo this doesnt need to be a call to `sync` we just need to read the current levels and not store them?